common/src/main/java/dev/latvian/mods/kubejs/item/custom/MultitoolItemJS.java

```diff
@@ -0,0 +1,179 @@
+        }
+        @Override
+        public Builder speedBaseline(float f) {
+            speedValues.clear();
+            speedValues.add(f);
+            return this;
+        }
+        protected void setValues() {
+            // TOD0: find a way to make this better / more efficient
+            // i hate floating-point jank
+            BigDecimal attack = new BigDecimal("0"), speed = attack;
+            for (final float f : attackValues) {
+                attack = attack.add(new BigDecimal(Float.toString(f)));
+            }
+            attack = attack.divide(new BigDecimal(Integer.toString(attackValues.size())), 3, CEILING);
+            attackDamageBaseline = attack.add(attack.multiply(BigDecimal.valueOf(.08)))
+                .setScale(1, CEILING).floatValue();
+
+            for (final float f : speedValues) {
+                speed = speed.add(new BigDecimal(Float.toString(f)));
+            }
+            speed = speed.divide(new BigDecimal(Integer.toString(speedValues.size())), 3, CEILING);
+            speedBaseline = speed.add(speed.multiply(BigDecimal.valueOf(.05)))
+                .setScale(1, CEILING).floatValue();
+        }
+        @Override
+        public Item createObject() {
+            setValues();
+            return new MultitoolItemJS(attackDamageBaseline, speedBaseline, toolTier, createItemProperties(), this);
+        }
+    }
+    public final Set<TagKey<Block>> mineableTags;
+    protected boolean isAxe, isHoe, isPickaxe, isShovel;
+    private final Multimap<ResourceLocation, AttributeModifier> attributes;
+    private boolean modified = false;
+    public MultitoolItemJS(float attack, float speed, MutableToolTier tier, Properties properties, Builder builder) {
+        super(attack, speed, tier, null, properties);
+        defaultModifiers = ArrayListMultimap.create(defaultModifiers);
+        mineableTags = Set.copyOf(builder.mineableTags);
+        isAxe = builder.isAxe;
+        isHoe = builder.isHoe;
+        isPickaxe = builder.isPickaxe;
+        isShovel = builder.isShovel;
+        attributes = builder.attributes;
+    }
+    public boolean isAxe() {
+        return isAxe;
+    }
+    public boolean isHoe() {
+        return isHoe;
+    }
+    public boolean isPickaxe() {
+        return isPickaxe;
+    }
+    public boolean isShovel() {
+        return isShovel;
+    }
+    @Override
+    public float getDestroySpeed(ItemStack itemStack, BlockState blockState) {
+        return isInMineables(blockState) ? speed : 1f;
+    }
+    @Override
+    public Multimap<Attribute, AttributeModifier> getDefaultAttributeModifiers(EquipmentSlot equipmentSlot) {
+        if (!modified) {
+            modified = true;
+            attributes.forEach((r, m) -> defaultModifiers.put(KubeJSRegistries.attributes().get(r), m));
+        }
+        return super.getDefaultAttributeModifiers(equipmentSlot);
+    }
+    @Override
+    // right-clicking on dirt will till, unless sneaking, then it will turn it into a path
```

Comment:
Is this consistent with other mods with multitools?

Reply:
im not sure, but idk how i would do it otherwise
(i couldn't find any paxel mods that can till and hoe at the same time)